expertise/create_dataset.py

```diff
@@ -85,17 +87,59 @@ def deduplicate_publications(self, publications):
                 deduplicated.append(pub)
 
         return deduplicated
+    
+    def get_pub_weight(self, pub=None, venueid=None, weight_specification=None):
+        if not venueid or not weight_specification:
+            return None
+            
+        # Get domain from either domain field or invitation prefix
+        domain = getattr(pub, 'domain', None)
+        if domain is None:
+            domain = pub.invitation.split('/-/')[0]  # API1 fallback to invitation
+        
+        # Return early on DBLP papers (venueid =/= domain) and non-accepted papers (domain not in venue_list)
+        if not (venueid == domain and domain in self.venue_list):
+            return None
+        ## Papers allowed: accepted papers from an OpenReview venue
+            
+        # Find matching weight specification
+        fallback_weight = None
+        for venue_spec in weight_specification:
+            if 'prefix' in venue_spec and venueid.startswith(venue_spec['prefix']):
+                return venue_spec['weight']
+            if 'value' in venue_spec and venueid == venue_spec['value']:
+                return venue_spec['weight']
+            if 'inOpenReview' in venue_spec:
```

Comment:
what does it mean "inOpenReview"?

Is there a spec defined somewhere about the weight_specification schema?

Reply:
Logic defined for new `articleSubmittedToOpenReview` key in discussion